src/test/groovy/life/qbic/datamodel/dtos/business/OfferIdSpec.groovy

```diff
@@ -20,4 +20,28 @@ class OfferIdSpec extends Specification {
         then:
         offerId.getRandomPart().equals("abc")
     }
+
+    def 'confirm equals method override works as intended'() {
+        given: "two offer id instances with the same content"
+        OfferId idOne = new OfferId("project", "random", "1")
+        OfferId idOther = new OfferId("project", "random", "1")
+
+        when: "we compare them"
+        def isEqual = idOne.equals(idOther)
+
+        then: "the comparison shall return true"
+        isEqual
+    }
+
+    def 'Equality check shall return false, if id conent differs'() {
+        given: "two offer id instances with the same content"
+        OfferId idOne = new OfferId("project", "random", "1")
+        OfferId idOther = new OfferId("project", "random", "2")
+
+        when: "we compare them"
+        def isEqual = idOne.equals(idOther)
+
+        then: "the comparison shall return true"
+        !isEqual
+    }
```

Comment:
This only checks inequality based on versions. I would suggest doing a slightly data-driven approach here.
```suggestion
    def 'Equality check shall return false, if id conent differs'() {
        given: "two offer id instances with the same content"
        OfferId idOne = new OfferId(conservedPart, randomPart, version)
        OfferId idOther = new OfferId(conservedPart2, randomPart2, version2)

        when: "we compare them"
        def isEqual = idOne.equals(idOther)

        then: "the comparison shall return false"
        !isEqual
        
        where: "constructor arguments differ for all possible fields"
        conservedPart | conservedPart2 | randomPart | randomPart2 | version | version2 
        "project" | "project" | "random" | "random" | "1" | "2"
        "project" | "project" | "random" | "random" | "2" | "1"
        "project" | "project2" | "random" | "random" | "3" | "3"
        "project2" | "project" | "random" | "random" | "3" | "3"
        "project3" | "project3" | "random1" | "random2" | "4" | "4"
        "project3" | "project3" | "random2" | "random1" | "4" | "4"   
        "project3" | "project4" | "random3" | "random4" | "5" | "6"
    }
```